addons/bus/models/bus.py

```diff
@@ -170,6 +173,15 @@ def loop(self):
                     for event in events:
                         event.set()
 
+    def wakeup_workers(self):
+        """
+        Wake up all http workers that are waiting for an event, useful
+        on server shutdown when they can't reveive anymore messages.
+        """
+        for events in self.channels.values():
+            for event in events:
+                event.set()
```

Comment:
Shouldn't you empty `self.channels`?

Reply:
Done by the code that is waiting for the event already